test/framework/universal_cluster.go

```diff
@@ -133,6 +133,7 @@ func (c *UniversalCluster) DeployKuma(mode core.CpMode, opt ...KumaDeploymentOpt
 	} else {
 		opt = append([]KumaDeploymentOption{WithEnvs(Config.KumaUniversalEnvVars)}, opt...)
 	}
+	c.opts = kumaDeploymentOptions{}
```

Comment:
Resetting the `opts` to allow `DeployKuma` to be run multiple times. Otherwise, the options will be populated in the second run.

Reply:
What if you change the way how w create a deployment of kuma? Now we store it as a `control-plane`. Maybe we need a list of control-planes? Also, why do we need a second run?

Reply:
Because I wanted to upgrade the CP in a test case, so it shares the existing cluster object while changing the controlPlane state.

There is still one control plane version in my case, so don't need to maintail multiple control plane objects.